formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -130,28 +130,30 @@ internal class FormulaManagerImpl<Input, State, Output>(
 
         var result: Evaluation<Output>? = null
         isRunning = true
-        var firstRun = true
-        while (result == null) {
-            val lastFrame = frame
-            val evaluationId = globalEvaluationId
-            val evaluation = if (firstRun) {
-                firstRun = false
-                evaluation(input, evaluationId)
-            } else if (lastFrame == null || isEvaluationNeeded(lastFrame.associatedEvaluationId)) {
-                evaluation(input, evaluationId)
-            } else {
-                lastFrame.evaluation
-            }
+        try {
```

Comment:
Wrapped this is a try/finally to keep isRunning in a good state. Not sure if it is completely necessary